### PR TITLE
chore(component-testing): add test for correct exit code

### PR DESCRIPTION
**Pull request by lmiller1990**
Add a test to validate that a certain number of tests have failed when running under component testing.

I believe I did this correctly - haven't done much in this area of the code base before, lmk if I missed something. The e2e test harness is neat.

### CODE REVIEW

File: packages/server/test/e2e/0_component_tests_spec.js [hunk 1]
```diff
@@ -54,4 +57,21 @@ describe('e2e component tests', () => {
       return snapshot('all tests results summary', runSummary)
     })
   })
+
+  it('fails and exits with correct code', function () {
+    return e2e.exec(this, {
+      project,
+      config: {
+        video: false,
+        experimentalComponentTesting: true,
```
Comment (elevatebart) [inline packages/server/test/e2e/0_component_tests_spec.js:66]: I thought that our flag was simply `componentTesting` ?

Comment (elevatebart) [reply packages/server/test/e2e/0_component_tests_spec.js:66]: Maybe this flag can be removed...

**Review by elevatebart — DISMISSED** [review]
There is a little comment I left but it's a real improvement overall.
Would you mind trying to make it fail and see that it does file when it should?

Breaking the cli.js run-ct command would probably do the trick ;)

File: packages/server/test/e2e/0_component_tests_spec.js (hunk 1 above)
Comment (lmiller1990) [reply packages/server/test/e2e/0_component_tests_spec.js:66]: I am actually unsure now - maybe this is not needed. I will look into it

**Review by JessicaSachs — APPROVED** [review]

**Pull request merged**